editor/script_editor_debugger.cpp

```diff
@@ -1132,10 +1132,15 @@ void ScriptEditorDebugger::_notification(int p_what) {
 				last_warning_count = warning_count;
 			}
 
-			if (connection.is_null()) {
-
-				if (server->is_connection_available()) {
-
+			if (server->is_connection_available()) {
+				if (connection.is_valid()) {
+					// We already have a valid connection.
+					// Disconnecting any new connecting client to prevent it from hanging.
+					Ref<StreamPeerTCP> conn = server->take_connection();
```

Comment:
This could just be `server->take_connection();` (going out of scope will cause the TCP stream to close), but I prefer the explicit code (while less performant). Let me know if I should change it.

Reply:
Maybe I should actually just do that and add it in the comment...